### PR TITLE
TF displays, show/hide all or individually

### PR DESCRIPTION
**Pull request by chapulina**
Addresses #6 and https://github.com/jstnhuang/ros-rviz/issues/21 .

pr2_tf

Note that this doesn't offer all the options which RViz does. Having a timeout could be good for performance in some cases.

### CODE REVIEW

File: ros-rviz-tf.html [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+        }
+
+        var that = this;
+        this._topic.subscribe(function(msg) {
+          that._onTf(that, msg);
+        });
+      },
+
+      _onTf: function(that, msg) {
+
```
Comment (jubeira) [inline ros-rviz-tf.html]: nit: extra space

File: ros-rviz-tf.html [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+        this.tfClient.subscribe(frame, function(tf) {
+          that._callback(that, frame, tf);
+        });
+      },
+
+      _toggleFrame: function(e) {
+        var frame = e.target.value;
+        var show = e.target.checked;
+
+        for (var i in this._tfs) {
```
Comment (jubeira) [inline ros-rviz-tf.html]: how about using [filter](https://www.vinta.com.br/blog/2015/javascript-lambda-and-arrow-functions/) on `_tfs`? Perhaps you can also get the index once you have the element?

File: ros-rviz-tf.html [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+              this._showTf(this._tfs[i], i);
+            }
+            else {
+              this._hideTf(this._tfs[i], i);
+            }
+            break;
+          }
+        }
+      },
+
```
Comment (jubeira) [inline ros-rviz-tf.html:181]: nit: extra line

**Review by jubeira — COMMENTED** [review]
Codewise looks good! I only left some minor comments.

The TF tree looks great as well. One super minor thing: when you zoom in, the axes are a bit thick compared to the arrows.
image
Perhaps it's worth changing the shaft radius?

File: ros-rviz-tf.html (hunk 1 above)
Comment (chapulina) [reply ros-rviz-tf.html]: Removed d5e07e6

File: ros-rviz-tf.html (hunk 3 above)
Comment (chapulina) [reply ros-rviz-tf.html:181]: Removed d5e07e6

File: ros-rviz-tf.html (hunk 2 above)
Comment (chapulina) [reply ros-rviz-tf.html]: Good idea, refactored several bits to use similar functions d5e07e6

**Comment by chapulina** [comment]
Thank you for the review, @jubeira !

Nice catch with the arrow heads. I really wanted to remove them, but `ROS3D.Axes` doesn't offer an API for that, so I just made them very small for now. In the future, we could either add the API there, or implement our custom class. Since TFs are usually many and small, I think it would be good for performance not to have the heads.

**Review by jubeira — APPROVED** [review]
LGTM!

**Pull request merged**